plutus-core/plutus-core/src/PlutusCore/Core/Instance/Pretty/Classic.hs

```diff
@@ -54,6 +54,10 @@ instance (PrettyClassicBy configName tyname, Pretty (SomeTypeIn uni), Pretty ann
         TyLam ann n k t    ->
             sexp "lam" (consAnnIf config ann
                 [prettyBy config n, prettyBy config k, prettyBy config t])
+        TySOP ann tyls    ->
+            sexp "sop" (consAnnIf config ann (fmap prettyTyl tyls))
+            where
+              prettyTyl tyl = brackets (sep (fmap (prettyBy config) tyl))
```

Comment:
Yes. Perhaps a bad choice? But I chose to do it like application, i.e. `[a b c]`.

Reply:
I just looked at things like

```
plutus-core/plutus-ir/src/PlutusIR/Compiler/Datatype.hs:SOPs: @mkPatternFunctorBody List = sop [] [a, List a]@
plutus-core/plutus-ir/src/PlutusIR/Compiler/Datatype.hs:        = fix (\(List :: * -> *) (a :: *) -> \(a :: *) -> sop [] [a, List a])
```

and was confused about the syntax. I don't mind separating types by spaces, but only because the classic syntax was doomed from the start :slightly_smiling_face:

Reply:
That's just me being inconsistent :upside_down_face: